Include/internal/pycore_weakref.h

```diff
@@ -8,49 +8,76 @@ extern "C" {
 #  error "this header requires Py_BUILD_CORE define"
 #endif
 
+#include "pycore_object.h"           // _Py_REF_IS_MERGED()
 #include "pycore_critical_section.h" // Py_BEGIN_CRITICAL_SECTION()
 
-static inline PyObject* _PyWeakref_GET_REF(PyObject *ref_obj) {
-    assert(PyWeakref_Check(ref_obj));
-    PyWeakReference *ref = _Py_CAST(PyWeakReference*, ref_obj);
-    PyObject *obj = ref->wr_object;
-
-    if (obj == Py_None) {
-        // clear_weakref() was called
-        return NULL;
+static inline int _is_dead(PyObject *obj)
+{
+    if (obj == NULL) {
```

Comment:
Is this check necessary? It doesn't look like `obj` should every be NULL because `wr_object` uses `Py_None` for dead objects instead of NULL.